File: Writer.m
function Writer(image,name)
ROI_x = [-0.5,0.5]*1e-3; %mm
ROI_y = [-0.5,0.5]*1e-3;
ROI_x_pix = round(2048*5/2 + ROI_x/(5.5e-6/5));
ROI_y_pix = round(2048*5/2 + ROI_y/(5.5e-6/5));
    I = sqrt(abs(image(ROI_x_pix(1):ROI_x_pix(2),ROI_y_pix(1):ROI_y_pix(2))));
    J = I / max(I(:));
    J = uint8(round(J * 255));
    J = 255 - J;
    imwrite(J, colormap(gray(256)), [name '.png'], 'BitDepth', 8);
end

File: cpp/Writer.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

using namespace cv;

void Writer(Mat image, std::string name)
{
    Mat I, J;
    image.convertTo(I, CV_32F);
    sqrt(abs(I), I);
    double max_I;
    minMaxLoc(I, NULL, &max_I);
    J = I / max_I;
    J.convertTo(J, CV_8UC1, 255, 0);
    bitwise_not(J, J);
    Mat inverted_image = 255 - image;
    imwrite(name + ".png", J);
}